Competition2018/src/main/java/competition/subsystems/elevator/commands/TrackCurrentUsageCommand.java

```diff
@@ -0,0 +1,61 @@
+package competition.subsystems.elevator.commands;
+
+import java.util.HashMap;
+
+import com.google.inject.Inject;
+
+import edu.wpi.first.wpilibj.Timer;
+import xbot.common.command.BaseCommand;
+import xbot.common.controls.actuators.XCANTalon;
+
+public class TrackCurrentUsageCommand extends BaseCommand {
+
+    XCANTalon talon;
+    double current;
+    double time;
+    double currentAveragingWindow;
+
+    HashMap<Double, Double> currentHistory;
+    
+    @Inject
+    public TrackCurrentUsageCommand(XCANTalon talon) {
+        this.talon = talon;
+        currentHistory = new HashMap<Double, Double>();
+        currentAveragingWindow = 1000;
+    }
+
+    @Override
+    public void initialize() {
+        
+    }
+
+    @Override
+    public void execute() {
+        current = talon.getOutputCurrent();
+        time = Timer.getFPGATimestamp();
+        currentHistory.put(time, current);
+        for (Double presentTime : currentHistory.keySet()) {
+            if (presentTime < time - currentAveragingWindow) {
+                currentHistory.remove(presentTime);
+            }
+        }
+    }
+
+    public double getAverageCurrent() {
+        double sum = 0;
+        for (Double presentTime : currentHistory.keySet()) {
```

Comment:
You can clean up loops like this using Streams; see here: https://stackoverflow.com/questions/10791568/calculating-average-of-an-array-list

Reply:
Good link! @tfrancisco , @veronayue , I recommend you should still finish up your code with the current implementation, and then we can go over Streams and how they can make your life easier.